charms/istio-ingressgateway/src/charm.py

```diff
@@ -232,5 +243,16 @@ def check_ca_root_cert(self, namespace):
         return config_map
 
 
+class CheckFailed(Exception):
+    """ Raise this exception if one of the checks in main fails. """
+
+    def __init__(self, msg, status_type=None):
+        super().__init__()
+
+        self.msg = msg
```

Comment:
Handle messages that are not strings (for example, passing exceptions directly).  This came up in some of our other charms.

```suggestion
        self.msg = str(msg)
```